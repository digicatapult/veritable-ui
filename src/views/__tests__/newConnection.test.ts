import { expect } from 'chai'
import { describe, it } from 'mocha'

import {
  successResponse,
  testErrorTargetBox,
  testSuccessTargetBox,
} from '../../models/__tests__/fixtures/companyHouseFixtures.js'
import NewConnectionTemplates from '../newConnection.js'

describe('NewConnectionTemplates', () => {
  describe('show form', () => {
    it('should render Error message with text error test', async () => {
      const templates = new NewConnectionTemplates()
      const rendered = await templates.companyEmptyTextBox({ errorMessage: 'error test' })
      expect(rendered).to.matchSnapshot()
    })

    it('should render a valid company in filled company test box', async () => {
      const templates = new NewConnectionTemplates()
      const rendered = await templates.companyFilledTextBox({ company: successResponse })
      expect(rendered).to.matchSnapshot()
    })

    it('should render form with a errormessage and invlaid response', async () => {
      const templates = new NewConnectionTemplates()
      const rendered = await templates.companyFormInput({ targetBox: testErrorTargetBox, formStage: 'form' })
      expect(rendered).to.matchSnapshot()
    })

    it('should render form with a valid repsponse', async () => {
      const templates = new NewConnectionTemplates()
      const rendered = await templates.companyFormInput({ targetBox: testSuccessTargetBox, formStage: 'form' })
      expect(rendered).to.matchSnapshot()
    })

    it('should render a confirmation page with given email and company number', async () => {
      const templates = new NewConnectionTemplates()
      const rendered = await templates.companyFormInput({
        targetBox: testSuccessTargetBox,
        formStage: 'confirmation',
        email: '123@123.com',
        companyNumber: successResponse.company_number,
      })
      expect(rendered).to.matchSnapshot()
    })

    it('should render a success response page with a single button to return to home', async () => {
      const templates = new NewConnectionTemplates()
      const rendered = await templates.companyFormInput({ targetBox: testSuccessTargetBox, formStage: 'success' })
      expect(rendered).to.matchSnapshot()
    })

    it('should a web page with the a form in an empty state', async () => {
      const templates = new NewConnectionTemplates()
      const rendered = await templates.formPage(testErrorTargetBox, 'form')
      expect(rendered).to.matchSnapshot()
    })

    it('should render a stepper html at stage 1', async () => {
      const templates = new NewConnectionTemplates()
      const rendered = await templates.stepper({ formStage: 'form' })
      expect(rendered).to.matchSnapshot()
    })

    it('should render a stepper html at stage 2', async () => {
      const templates = new NewConnectionTemplates()
      const rendered = await templates.stepper({ formStage: 'confirmation' })
      expect(rendered).to.matchSnapshot()
    })

    it('should render a stepper html at stage 3', async () => {
      const templates = new NewConnectionTemplates()
      const rendered = await templates.stepper({ formStage: 'success' })
      expect(rendered).to.matchSnapshot()
    })
  })
})
