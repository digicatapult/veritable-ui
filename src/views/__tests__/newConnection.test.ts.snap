// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewConnectionTemplates show form should a web page with the a form in an empty state 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx.org/ext/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>Veritable - New Connection</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><nav class=\\"flex-page side-bar\\"><img class=\\"side-bar logo-container\\" src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"#category\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/category.svg&#34;);\\"></a><a title=\\"connections\\" href=\\"/connection\\" class=\\"side-bar icon active\\" style=\\"background-image:url(&#34;/public/images/connection.svg&#34;);\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/folder.svg&#34;);\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/notification.svg&#34;);\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/setting.svg&#34;);\\"></a></nav><div class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\">New Connection</h1><div class=\\"header nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><span class=\\"url-separator\\">/</span><a title=\\"New Connection\\" href=\\"/connection/new\\">New Connection</a></div></div><div class=\\"content main\\"><div class=\\"connections header\\"><span>Invite New Connection</span></div><form id=\\"new-connection-form\\" class=\\"new-connection\\" hx-post=\\"/connection/new/submit\\" hx-swap=\\"outerHTML\\" hx-vals=\\"{&#34;formStage&#34;: &#34;form&#34;}\\"><div class=\\"new-connection col\\"><div>-----Step 1</div><input id=\\"company-number-input\\" class=\\"new connection\\" name=\\"companyNumber\\" placeholder=\\"Company House Number\\" required hx-get=\\"/connection/new/verify-company\\" hx-trigger=\\"keyup changed delay:200ms, change\\" hx-target=\\"#text-box-target\\" hx-select=\\"#text-box-target\\" pattern=\\"^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\\\\d{2})\\\\d{6})|((IP|SP|RS)[A-Z\\\\d]{6})|(SL\\\\d{5}[\\\\dA]))$\\" minlength=\\"8\\" maxlength=\\"8\\" oninput=\\"this.reportValidity()\\" type=\\"text\\"/><input id=\\"email\\" class=\\"new-connection\\" placeholder=\\"Connection's Email Address\\" name=\\"email\\" type=\\"text\\"/><div id=\\"confirmation-page\\" style=\\"display:none;\\"><p>Please confirm the details of the connection before sending</p><p>Company House Number: false</p><p>Email Address: false</p></div><div id=\\"success-page\\" style=\\"display:none;\\"><p>Your connection invitation has been sent. Please wait for their verification. As the post may take 2-3 days to arrive, please wait for their verification and keep updated by viewing the verification status.</p></div><a class=\\"button\\" href=\\"/connection\\" style=\\"display:block;\\">Cancel</a><button style=\\"display:none\\" class=\\"button undefined undefined\\" type=\\"submit\\" name=\\"submitButton\\" value=\\"Back\\"><span class=\\"button text undefined\\">Back</span></button><button style=\\"display:block\\" class=\\"button undefined undefined\\" type=\\"submit\\" name=\\"submitButton\\" value=\\"Continue\\"><span class=\\"button text undefined\\">Continue</span></button></div><div id=\\"text-box-target\\" class=\\"new-connection col\\" style=\\"display:block;\\">This is a test error message</div></form></div></div></body></html>"`;

exports[`NewConnectionTemplates show form should render Error message with text error test 1`] = `"error test"`;

exports[`NewConnectionTemplates show form should render a confirmation page with given email and company number 1`] = `"<form id=\\"new-connection-form\\" class=\\"new-connection\\" hx-post=\\"/connection/new/submit\\" hx-swap=\\"outerHTML\\" hx-vals=\\"{&#34;formStage&#34;: &#34;confirmation&#34;}\\"><div class=\\"new-connection col\\"><div>----------Step 2</div><input id=\\"company-number-input\\" class=\\"new connection\\" name=\\"companyNumber\\" placeholder=\\"Company House Number\\" required hx-get=\\"/connection/new/verify-company\\" hx-trigger=\\"keyup changed delay:200ms, change\\" hx-target=\\"#text-box-target\\" hx-select=\\"#text-box-target\\" pattern=\\"^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\\\\d{2})\\\\d{6})|((IP|SP|RS)[A-Z\\\\d]{6})|(SL\\\\d{5}[\\\\dA]))$\\" minlength=\\"8\\" maxlength=\\"8\\" oninput=\\"this.reportValidity()\\" value=\\"10592650\\" type=\\"hidden\\"/><input id=\\"email\\" class=\\"new-connection\\" placeholder=\\"Connection's Email Address\\" name=\\"email\\" value=\\"123@123.com\\" type=\\"hidden\\"/><div id=\\"confirmation-page\\" style=\\"display:block;\\"><p>Please confirm the details of the connection before sending</p><p>Company House Number: 10592650</p><p>Email Address: 123@123.com</p></div><div id=\\"success-page\\" style=\\"display:none;\\"><p>Your connection invitation has been sent. Please wait for their verification. As the post may take 2-3 days to arrive, please wait for their verification and keep updated by viewing the verification status.</p></div><a class=\\"button\\" href=\\"/connection\\" style=\\"display:none;\\">Back To Home</a><button style=\\"display:block\\" class=\\"button undefined undefined\\" type=\\"submit\\" name=\\"submitButton\\" value=\\"Back\\"><span class=\\"button text undefined\\">Back</span></button><button style=\\"display:block\\" class=\\"button undefined undefined\\" type=\\"submit\\" name=\\"submitButton\\" value=\\"Submit\\"><span class=\\"button text undefined\\">Submit</span></button></div><div id=\\"text-box-target\\" class=\\"new-connection col\\" style=\\"display:none;\\"><p>SMH IOT SOLUTIONS LTD</p><p>51 Mornington Crescent</p><p>Hounslow</p><p>TW5 9ST</p><p>United Kingdom</p><p>dissolved</p></div></form>"`;

exports[`NewConnectionTemplates show form should render a stepper html at stage 1 1`] = `"<div>-----Step 1</div>"`;

exports[`NewConnectionTemplates show form should render a stepper html at stage 2 1`] = `"<div>----------Step 2</div>"`;

exports[`NewConnectionTemplates show form should render a stepper html at stage 3 1`] = `"<div>---------------Step 3</div>"`;

exports[`NewConnectionTemplates show form should render a success response page with a single button to return to home 1`] = `"<form id=\\"new-connection-form\\" class=\\"new-connection\\" hx-post=\\"/connection/new/submit\\" hx-swap=\\"outerHTML\\" hx-vals=\\"{&#34;formStage&#34;: &#34;success&#34;}\\"><div class=\\"new-connection col\\"><div>---------------Step 3</div><input id=\\"company-number-input\\" class=\\"new connection\\" name=\\"companyNumber\\" placeholder=\\"Company House Number\\" required hx-get=\\"/connection/new/verify-company\\" hx-trigger=\\"keyup changed delay:200ms, change\\" hx-target=\\"#text-box-target\\" hx-select=\\"#text-box-target\\" pattern=\\"^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\\\\d{2})\\\\d{6})|((IP|SP|RS)[A-Z\\\\d]{6})|(SL\\\\d{5}[\\\\dA]))$\\" minlength=\\"8\\" maxlength=\\"8\\" oninput=\\"this.reportValidity()\\" type=\\"hidden\\"/><input id=\\"email\\" class=\\"new-connection\\" placeholder=\\"Connection's Email Address\\" name=\\"email\\" type=\\"hidden\\"/><div id=\\"confirmation-page\\" style=\\"display:none;\\"><p>Please confirm the details of the connection before sending</p><p>Company House Number: 10592650</p><p>Email Address: </p></div><div id=\\"success-page\\" style=\\"display:block;\\"><p>Your connection invitation has been sent. Please wait for their verification. As the post may take 2-3 days to arrive, please wait for their verification and keep updated by viewing the verification status.</p></div><a class=\\"button\\" href=\\"/connection\\" style=\\"display:block;\\">Back To Home</a><button style=\\"display:none\\" class=\\"button undefined undefined\\" type=\\"submit\\" name=\\"submitButton\\" value=\\"Back\\"><span class=\\"button text undefined\\">Back</span></button><button style=\\"display:none\\" class=\\"button undefined undefined\\" type=\\"submit\\" name=\\"submitButton\\" value=\\"Submit\\"><span class=\\"button text undefined\\">Submit</span></button></div><div id=\\"text-box-target\\" class=\\"new-connection col\\" style=\\"display:none;\\"><p>SMH IOT SOLUTIONS LTD</p><p>51 Mornington Crescent</p><p>Hounslow</p><p>TW5 9ST</p><p>United Kingdom</p><p>dissolved</p></div></form>"`;

exports[`NewConnectionTemplates show form should render a valid company in filled company test box 1`] = `"<p>SMH IOT SOLUTIONS LTD</p><p>51 Mornington Crescent</p><p>Hounslow</p><p>TW5 9ST</p><p>United Kingdom</p><p>dissolved</p>"`;

exports[`NewConnectionTemplates show form should render form with a errormessage and invlaid response 1`] = `"<form id=\\"new-connection-form\\" class=\\"new-connection\\" hx-post=\\"/connection/new/submit\\" hx-swap=\\"outerHTML\\" hx-vals=\\"{&#34;formStage&#34;: &#34;form&#34;}\\"><div class=\\"new-connection col\\"><div>-----Step 1</div><input id=\\"company-number-input\\" class=\\"new connection\\" name=\\"companyNumber\\" placeholder=\\"Company House Number\\" required hx-get=\\"/connection/new/verify-company\\" hx-trigger=\\"keyup changed delay:200ms, change\\" hx-target=\\"#text-box-target\\" hx-select=\\"#text-box-target\\" pattern=\\"^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\\\\d{2})\\\\d{6})|((IP|SP|RS)[A-Z\\\\d]{6})|(SL\\\\d{5}[\\\\dA]))$\\" minlength=\\"8\\" maxlength=\\"8\\" oninput=\\"this.reportValidity()\\" type=\\"text\\"/><input id=\\"email\\" class=\\"new-connection\\" placeholder=\\"Connection's Email Address\\" name=\\"email\\" type=\\"text\\"/><div id=\\"confirmation-page\\" style=\\"display:none;\\"><p>Please confirm the details of the connection before sending</p><p>Company House Number: false</p><p>Email Address: false</p></div><div id=\\"success-page\\" style=\\"display:none;\\"><p>Your connection invitation has been sent. Please wait for their verification. As the post may take 2-3 days to arrive, please wait for their verification and keep updated by viewing the verification status.</p></div><a class=\\"button\\" href=\\"/connection\\" style=\\"display:block;\\">Cancel</a><button style=\\"display:none\\" class=\\"button undefined undefined\\" type=\\"submit\\" name=\\"submitButton\\" value=\\"Back\\"><span class=\\"button text undefined\\">Back</span></button><button style=\\"display:block\\" class=\\"button undefined undefined\\" type=\\"submit\\" name=\\"submitButton\\" value=\\"Continue\\"><span class=\\"button text undefined\\">Continue</span></button></div><div id=\\"text-box-target\\" class=\\"new-connection col\\" style=\\"display:block;\\">This is a test error message</div></form>"`;

exports[`NewConnectionTemplates show form should render form with a valid repsponse 1`] = `"<form id=\\"new-connection-form\\" class=\\"new-connection\\" hx-post=\\"/connection/new/submit\\" hx-swap=\\"outerHTML\\" hx-vals=\\"{&#34;formStage&#34;: &#34;form&#34;}\\"><div class=\\"new-connection col\\"><div>-----Step 1</div><input id=\\"company-number-input\\" class=\\"new connection\\" name=\\"companyNumber\\" placeholder=\\"Company House Number\\" required hx-get=\\"/connection/new/verify-company\\" hx-trigger=\\"keyup changed delay:200ms, change\\" hx-target=\\"#text-box-target\\" hx-select=\\"#text-box-target\\" pattern=\\"^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\\\\d{2})\\\\d{6})|((IP|SP|RS)[A-Z\\\\d]{6})|(SL\\\\d{5}[\\\\dA]))$\\" minlength=\\"8\\" maxlength=\\"8\\" oninput=\\"this.reportValidity()\\" type=\\"text\\"/><input id=\\"email\\" class=\\"new-connection\\" placeholder=\\"Connection's Email Address\\" name=\\"email\\" type=\\"text\\"/><div id=\\"confirmation-page\\" style=\\"display:none;\\"><p>Please confirm the details of the connection before sending</p><p>Company House Number: 10592650</p><p>Email Address: </p></div><div id=\\"success-page\\" style=\\"display:none;\\"><p>Your connection invitation has been sent. Please wait for their verification. As the post may take 2-3 days to arrive, please wait for their verification and keep updated by viewing the verification status.</p></div><a class=\\"button\\" href=\\"/connection\\" style=\\"display:block;\\">Cancel</a><button style=\\"display:none\\" class=\\"button undefined undefined\\" type=\\"submit\\" name=\\"submitButton\\" value=\\"Back\\"><span class=\\"button text undefined\\">Back</span></button><button style=\\"display:block\\" class=\\"button undefined undefined\\" type=\\"submit\\" name=\\"submitButton\\" value=\\"Continue\\"><span class=\\"button text undefined\\">Continue</span></button></div><div id=\\"text-box-target\\" class=\\"new-connection col\\" style=\\"display:block;\\"><p>SMH IOT SOLUTIONS LTD</p><p>51 Mornington Crescent</p><p>Hounslow</p><p>TW5 9ST</p><p>United Kingdom</p><p>dissolved</p></div></form>"`;
